sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs

```diff
@@ -205,6 +200,57 @@ public void IgnoresConstructorWhenCredentialsNull()
             Assert.AreSame(clientOptions, client.Options);
         }
 
+        [Test]
+        public void IgnoresNonTokenCredentialConstructors()
+        {
+            IConfiguration configuration = GetConfiguration(
+                new KeyValuePair<string, string>("uri", "http://localhost"),
+                new KeyValuePair<string, string>("credential", "managedidentity"),
+                new KeyValuePair<string, string>("clientId", "secret")
+            );
+
+            var clientOptions = new TestClientOptions();
+            var client = (TestClientWithCredentials)ClientFactory.CreateClient(typeof(TestClientWithCredentials), typeof(TestClientOptions), clientOptions, configuration, new DefaultAzureCredential());
+
+            Assert.AreEqual("http://localhost/", client.Uri.ToString());
+            Assert.AreSame(clientOptions, client.Options);
+            Assert.NotNull(client.Credential);
+        }
+
+        [Test]
+        public void CanUseAzureKeyCredential()
```

Comment:
this might not work well with https://github.com/Azure/azure-sdk-for-net/blob/2bbf6abc21fea1ae2c96da02c4c83e123c8cb193/sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs#L128

I wonder if we should add AzureKeyCredential to storage or haven an extension point to handle custom credential types. I recall that when we discussed sas credential consensus was to leave storage as is and introduce key credential from core elsewhere.

Reply:
>this might not work well with

Why not?

Reply:
I was guessing that when this sees "credential:key"  it somehows think that AzureKeyCredential should be resolved. Is this capable of reflectively create StorageSharedKeyCredential ?

Reply:
Yep, it looks at ctor parameters. So if someone provides the right parameters for StorageSharedKeyCredential we'll create it.